test/widgets/user_test.dart

```diff
@@ -78,5 +83,51 @@ void main() {
       check(await actualUrl(tester, avatarUrl)).isNull();
       debugNetworkImageHttpClientProvider = null;
     });
+
+    testWidgets('shows placeholder when image URL gives error', (WidgetTester tester) async {
+      addTearDown(testBinding.reset);
+      prepareBoringImageHttpClient(success: false);
+      await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot());
+      final store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+
+      final badUser = eg.user(avatarUrl: 'https://zulip.com/avatarinvalid.png');
+      await store.addUser(badUser);
+
+      await tester.pumpWidget(
+         TestZulipApp(accountId: eg.selfAccount.id,
+           child: AvatarImage(userId: badUser.userId, size: 30)));
```

Comment:
Part of https://github.com/zulip/zulip-flutter/pull/1882#discussion_r2548238195 still isn't done; bump:

```suggestion
      await tester.pumpWidget(
        TestZulipApp(accountId: eg.selfAccount.id,
          child: AvatarImage(userId: badUser.userId, size: 30)));
```